src/Microsoft.Performance.SDK.Runtime/Options/Serialization/Loading/StreamPluginOptionsLoader.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Performance.SDK.Runtime.Options.Serialization.DTO;
+
+namespace Microsoft.Performance.SDK.Runtime.Options.Serialization.Loading;
+
+/// <summary>
+///     Base class for <see cref="IPluginOptionsLoader"/> classes which load from a stream.
+/// </summary>
+public abstract class StreamPluginOptionsLoader
+    : IPluginOptionsLoader
+{
+    private readonly bool closeStreamOnWrite;
+
+    /// <summary>
+    ///     Initializes a new instance of the <see cref="StreamPluginOptionsLoader"/> class.
+    /// </summary>
+    /// <param name="closeStreamOnWrite">
+    ///     Whether to dispose of the stream returned by <see cref="GetStream"/> at the end of a call to
+    ///     <see cref="TryLoadAsync"/>.
+    /// </param>
+    protected StreamPluginOptionsLoader(bool closeStreamOnWrite)
+    {
+        this.closeStreamOnWrite = closeStreamOnWrite;
+    }
+
+    /// <inheritdoc />
+    public async Task<PluginOptionsDto> TryLoadAsync()
+    {
+        var jsonSerializerOptions = new JsonSerializerOptions
+        {
+            WriteIndented = true,
+            PropertyNameCaseInsensitive = true,
+        };
+
+        var stream = GetStream();
+        try
+        {
+            return await JsonSerializer.DeserializeAsync<PluginOptionsDto>(stream, jsonSerializerOptions);
+        }
+        catch
+        {
+            return null;
```

Comment:
>return null; [](http://example.com/codeflow?start=12&length=12)

do we want any sort of logging for this failure case?

Reply:
Yes - my next PR that I have stacked on this branch will add logging.